runhouse/main.py

```diff
@@ -576,6 +592,95 @@ def status(
     _print_status(cluster_status, current_cluster)
 
 
+async def aget_clusters_from_den():
+    httpx_client = httpx.AsyncClient()
+
+    get_clusters_params = {"resource_type": "cluster", "folder": rns_client.username}
+    clusters_in_den_resp = await httpx_client.get(
+        f"{rns_client.api_server_url}/resource",
+        params=get_clusters_params,
+        headers=rns_client.request_headers(),
+    )
+
+    return clusters_in_den_resp
+
+
+def get_clusters_from_den():
+    return asyncio.run(aget_clusters_from_den())
+
+
+@cluster_app.command("list")
+def cluster_list():
+    """Load Runhouse clusters"""
+    import sky
+
+    # logged out case
+    if not rh.configs.token:
+        # TODO [SB]: adjust msg formatting (coloring etc)
+        sky_cli_command_formatted = f"{italic_bold_ansi}sky status -r{reset_format}"  # will be printed bold and italic
+        console.print(
+            f"Runhouse token required to view all clusters. Please run `runhouse login` to load your token. To view on-demand clusters run {sky_cli_command_formatted}"
+        )
+        return
+
+    sky_clusters = sky.status(refresh=True)
+
+    clusters_in_den_resp = get_clusters_from_den()
+
+    if clusters_in_den_resp.status_code != 200:
+        logger.error(
+            f"Failed to load clusters from Den for username: {rns_client.username}"
+        )
+        clusters_in_den = []
+    else:
+        clusters_in_den = clusters_in_den_resp.json().get("data")
+
+    clusters_in_den_names = [cluster.get("name") for cluster in clusters_in_den]
+
+    if not sky_clusters and not clusters_in_den:
+        console.print("No existing clusters.")
+
+    if sky_clusters:
+        # getting the on-demand clusters that are not saved in den.
+        sky_clusters = [
+            cluster
+            for cluster in sky_clusters
+            if f'/{rns_client.username}/{cluster.get("name")}'
+            not in clusters_in_den_names
+        ]
+
+    total_clusters = len(clusters_in_den)
+    table_title = f"[bold cyan]{rns_client.username}'s Clusters (Total: {total_clusters})[/bold cyan]"
```

Comment:
wondering if instead of just giving a total, we also show some sort of high level breakdown for number of clusters that are running vs not running (in the case that no filters are specified).. wdyt?

Reply:
ah nevermind, i see you did that in the following PR :)